homeassistant/bootstrap.py

```diff
@@ -388,10 +394,12 @@ def async_from_config_dict(config: Dict[str, Any],
         yield from hass.loop.run_in_executor(None, loader.prepare, hass)
 
     # Make a copy because we are mutating it.
-    # Convert it to defaultdict so components can always have config dict
+    # Use OrderedDict in case original one was one.
     # Convert values to dictionaries if they are None
-    config = defaultdict(
-        dict, {key: value or {} for key, value in config.items()})
+    new_config = OrderedDict()
```

Comment:
I always thought that we setup everything with an ordered dict but looks like I was wrong. Instead we converted everything to a normal dictionary, making each setup unpredictable.